### PR TITLE
CB-5944 List api methods should be filtered correctly based on Crn

### PR DESCRIPTION
**Pull request by horadla23**

### CODE REVIEW

**Comment by foldik** [comment]
Good stuff overall.

File: auth-connector/src/main/java/com/sequenceiq/cloudbreak/auth/altus/GrpcUmsClient.java [hunk 1]
```diff
@@ -379,13 +379,15 @@ public boolean checkRight(String actorCrn, String userCrn, String right, Optiona
...
         List<AuthorizationProto.RightCheck> rightChecks = Lists.newArrayList();
-        rightCheckMap.entrySet().stream().forEach(entry -> rightChecks.add(AuthorizationProto.RightCheck.newBuilder()
-                .setResource(entry.getKey())
-                .setRight(entry.getValue())
+        resources.stream().forEach(resource -> rightChecks.add(AuthorizationProto.RightCheck.newBuilder()
+                .setResource(resource)
+                .setRight(right)
                 .build()));
-        return hasRights(actorCrn, memberCrn, rightChecks, requestId);
+        List<Boolean> result = hasRights(actorCrn, memberCrn, rightChecks, requestId);
```
[the remark this answers is not in the document]
Comment (horadla23) [reply auth-connector/src/main/java/com/sequenceiq/cloudbreak/auth/altus/GrpcUmsClient.java]: sure

File: auth-connector/src/main/java/com/sequenceiq/cloudbreak/auth/altus/GrpcUmsClient.java [hunk 2]
```diff
@@ -379,13 +379,15 @@ public boolean checkRight(String actorCrn, String userCrn, String right, Optiona
         }
     }
 
-    public List<Boolean> hasRights(String actorCrn, String memberCrn, Map<String, String> rightCheckMap, Optional<String> requestId) {
+    public Map<String, Boolean> hasRights(String actorCrn, String memberCrn, List<String> resources, String right, Optional<String> requestId) {
         List<AuthorizationProto.RightCheck> rightChecks = Lists.newArrayList();
```
[the remark this answers is not in the document]
Comment (horadla23) [reply auth-connector/src/main/java/com/sequenceiq/cloudbreak/auth/altus/GrpcUmsClient.java]: ok, I do not know why I didn't use for the first time :)

File: authorization-common/src/main/java/com/sequenceiq/authorization/service/ListPermissionChecker.java [hunk 3]
```diff
@@ -0,0 +1,117 @@
...
+            ProceedingJoinPoint proceedingJoinPoint, MethodSignature methodSignature, long startTime) {
+        FilterListBasedOnPermissions methodAnnotation = (FilterListBasedOnPermissions) rawMethodAnnotation;
+        AuthorizationResourceAction action = methodAnnotation.action();
+        List<String> allResourceCrns = resourceBasedCrnProviderMap.get(resourceType).getResourceCrnsInAccount();
+        List<String> filteredResourceCrns = commonPermissionCheckingUtils.getPermissionsForUserOnResources(resourceType, action, userCrn, allResourceCrns)
+                .entrySet()
+                .stream()
+                .filter(Map.Entry::getValue)
+                .map(Map.Entry::getKey)
+                .collect(Collectors.toList());
```
[the remark this answers is not in the document]
Comment (horadla23) [reply authorization-common/src/main/java/com/sequenceiq/authorization/service/ListPermissionChecker.java]: ok

**Comment by horadla23** [comment]
> I think the implementation is understandable.
> My only idea was that if this Logic could benefit from an actual higher level test with SpringbootTest. I'm not adding this as a requirement for this PR, but we should definitely think about that.

I agree, I created this jira for it: https://jira.cloudera.com/browse/CB-6274

**Pull request merged**